### ISSUE TITLE
Create a functional search bar with autocompletion for game names

### ISSUE THREAD

**Issue by rakrusch** [uchicago-cs/chigame#95]
Add a search bar that will autocomplete a user's query to search for games registered in ChiGame. We will need to wait on the user stories for searching to be completed in issue #90. The research for how to create a functional search bar was done in issue #91. The research for how to implement autocomplete was done in issue #94. This issue involves integrating that knowledge into one search bar with autocomplete. This search bar may share functionality with the users team. (Edited by Patrick)

**Comment by willcravitz** [comment]
We need to reach out to the users team (#89) to decide if we will share one search bar that lives in the navbar or if we will have our own distinct search that lives on the browse game page.

**Comment by ptlin25** [comment]
Created branch `game-management/autocomplete` for this issue

**Comment by ptlin25** [comment]
An initial version of the autocomplete feature has been added to search bar. Currently testing that it works

**Comment by ptlin25** [comment]
I changed the search bar's type to be HTML search input type instead of a standard text input. The autocomplete feature only fills in game names if the query type is Games.

**Comment by willcravitz** [comment]
I've been testing this and it works great. Here's an example:

![Image](https://github.com/uchicago-cs/chigame/assets/93291811/3470e53b-af7a-43ff-9860-8a8a6677e134)

But I loaded in the large game fixture, and when I type in a single letter, I get way too many results:

![Image](https://github.com/uchicago-cs/chigame/assets/93291811/91a7a1c7-9792-4ea2-b2cd-61c02f935e5b)

This could be worth opening up a new issue to address. We can either
1. Only show the autocomplete dropdown after a certain number of characters have been entered.
2. Limit the maximum number of results displayed in the dropdown.

**Comment by ptlin25** [comment]
Thanks for catching this. I just pushed a commit to address this issue by limiting the number of autocomplete results displayed to 10. Please take a look to help me test it and see if my comments do a good enough job explaining how it works

**Comment by ptlin25** [comment]
This task successfully added autocompletion for game names to the search bar in the navigation bar. When the user is in the "Games" mode of the search bar, the search bar will display the first 10 autocompletion results. In other words, when a user types something into the search bar, up to 10 game names that match the input will be displayed as tags underneath the search bar. These tags can be clicked on to redirect users to the game's detail page. 

Contributions:
Patrick: Coding the autocomplete feature
Will and Ryan: Testing and feedback

**Comment by majorsylvie** [comment]
Issue Score: Excellent

Comments:
This is so cool! 
Awesome work @ptlin25 on implementing this feature.
Great closing comment and summary! (appreciate you specifically saying you edited ryan's comment)

@willcravitz thank you for your very good updates. Images help a lot for an issue like this.

Great work y'all!
